samples/mpu/mem_domain_apis_test/src/main.c

```diff
@@ -26,13 +26,23 @@ struct k_thread app_thread_id[3];
 struct k_mem_domain app_domain[2];
 
 /* the start address of the MPU region needs to align with its size */
+#ifdef CONFIG_ARM
 u8_t __aligned(32) app0_buf[32];
 u8_t __aligned(32) app1_buf[32];
+#else
+u8_t __aligned(4096) app0_buf[4096];
```

Comment:
Can we have a `K_MEM_PARTITION_ALIGNED` (or something of the sort) that each platform specifies with the correct page size?

Reply:
It's hard to capture this in a completely generic way.
On x86, memory is configured at 4K page granularity, the regions have to be 4K aligned but the size can be any multiple of 4K.
With ARM MPU, a region can be of any power-of-two size starting at 32 bytes, and must be aligned to its size.
Other MPUs may be more permissive.
This sample needs to be rewritten as a test case anyway as it doesn't actually prove that anything works in terms that CI can understand.

Reply:
Having said this, one cheap way to get rid of the ifdef is just make the size 4096, aligned to 4096, that ought to work everywhere.